### PR TITLE
feat/authentication loginregister

### LINKED ISSUE #13: Authentication: Login/Register
- [x] create in typeDefs, resolvers, and services Authentication as if Authentication was a data model.
- [x] Create Login and Register mutations.
- [x] Make sure both of these upon success are able to set "auth_token" in cookies with the JWT token. Make sure the cookie is samesite: true, secure, httponly.
- Your login mutation should take in an email and password, check if the user exists and compare the password hashes.
- Your register mutation should take in email and password, check if the user exists and fail if they do exist. If not then hash the password and store it.

### PR DESCRIPTION
**Pull request by SerpentSnek**

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -7,39 +7,45 @@ datasource db {
...
-}
+// model Playlist {
+//   id        String           @id @default(uuid())
+//   name      String
+//   user      User             @relation(fields: [userId], references: [id], onDelete: Cascade)
+//   userId    String
+//   songs     PlaylistToSong[]
+//   createdAt DateTime         @default(now())
+//   updatedAt DateTime         @updatedAt
+// }
```
Comment (Austin2Shih) [inline prisma/schema.prisma]: no need to comment these out, if you just run the schema generate, it should clear them out for you. Same for anything else in the schema.prisma file

File: src/resolvers/index.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
-import User from "./User";
-import Playlist from "./Playlist";
-import Song from "./Song";
+// import User from "./User";
```
Comment (Austin2Shih) [inline src/resolvers/index.ts]: If you're going to remove this, just remove it completely, no need to keep it as a comment

File: src/services/Authentications.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  // READ
+  static async findEmail({ email }) {
+    try {
+      return prisma.authentication.findFirstOrThrow({
+        where: {
+          email,
+        },
+      });
+    } catch (e) {
+      throw new Error("something went wrong");
```
Comment (Austin2Shih) [inline src/services/Authentications.ts]: More explicit error messages perhaps. Only if e contains the error info of course.

**Review by Austin2Shih — CHANGES_REQUESTED** [review]
Mostly looks good, take a look at the code comments for more info on what to change. Also, update your branch to main when you're done with changes again.

File: prisma/models/.config.prisma [hunk 4]
```diff
@@ -5,4 +5,4 @@ generator client {
 datasource db {
   provider = "postgresql"
   url      = env("DATABASE_URL")
-}
\ No newline at end of file
+}
```
Comment (Austin2Shih) [inline prisma/models/.config.prisma:8]: maybe remove unnecessary changes

File: prisma/schema.prisma [hunk 5]
```diff
@@ -7,52 +7,7 @@ datasource db {
...
-  playlist   Playlist @relation(fields: [playlistId], references: [id], onDelete: Cascade)
-  songId     String
-  song       Song     @relation(fields: [songId], references: [id], onDelete: Cascade)
-  createdAt  DateTime @default(now())
-  updatedAt  DateTime @updatedAt
-
-  @@id(name: "id", fields: [playlistId, songId])
-}
-
-model Products {
```
Comment (Austin2Shih) [inline prisma/schema.prisma]: keep this in :0

File: src/resolvers/index.ts [hunk 6]
```diff
@@ -1,6 +1,3 @@
-import User from "./User";
-import Playlist from "./Playlist";
-import Song from "./Song";
-import Products from "./Products";
```
Comment (Austin2Shih) [inline src/resolvers/index.ts:4]: products needs to stay

File: src/resolvers/index.ts [hunk 7]
```diff
@@ -1,6 +1,3 @@
...
+import Authentication from "./Authentication";
 
-export default [User, Playlist, Products, Song];
```
Comment (Austin2Shih) [inline src/resolvers/index.ts:6]: products

File: src/typeDefs/index.ts [hunk 8]
```diff
@@ -1,6 +1,3 @@
-import User from "./User.js";
-import Playlist from "./Playlist.js";
-import Song from "./Song.js";
-import Products from "./Products.js";
```
Comment (Austin2Shih) [inline src/typeDefs/index.ts:4]: products

**Review by Austin2Shih — CHANGES_REQUESTED** [review]

File: prisma/models/.config.prisma (hunk 4 above)
Comment (SerpentSnek) [reply prisma/models/.config.prisma:8]: i have no record of this commit showing up on my branch, and i don't recall touching .config.prisma recently.
i'll try to avoid useless diffs like this in the future nonetheless.

**Review by Austin2Shih — APPROVED** [review]
looks good!
